src/veins/modules/mobility/traci/subscriptionManagement/TraCISubscriptionManager.h

```diff
@@ -0,0 +1,218 @@
+//
+// Copyright (C) 2006-2017 Christoph Sommer <sommer@ccs-labs.org>
+//
+// Documentation for these modules is at http://veins.car2x.org/
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+//
+
+#ifndef SRC_VEINS_MODULES_MOBILITY_TRACI_SUBSCRIPTIONMANAGEMENT_TRACISUBSCRIPTIONMANAGER_H_
+#define SRC_VEINS_MODULES_MOBILITY_TRACI_SUBSCRIPTIONMANAGEMENT_TRACISUBSCRIPTIONMANAGER_H_
+
+#include "veins/modules/mobility/traci/subscriptionManagement/PersonSubscriptionManager.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/SimulationSubscriptionManager.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/TraCIPerson.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/TraCITrafficLight.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/TraCIVehicle.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/TrafficLightSubscriptionManager.h"
+#include "veins/modules/mobility/traci/subscriptionManagement/VehicleSubscriptionManager.h"
+#include "veins/modules/mobility/traci/TraCIBuffer.h"
+#include "veins/modules/mobility/traci/TraCIConnection.h"
+#include "veins/modules/mobility/traci/TraCICommandInterface.h"
+
+namespace veins {
+namespace TraCISubscriptionManagement {
+
+/**
+ * @class TraCISubscriptionManager
+ *
+ * This manages all the subscriptions that were made and provides an API
+ * to the underlying subscription managers and request their results.
+ *
+ * @author Nico Dassler <dassler@hm.edu>
+ */
+class TraCISubscriptionManager {
+
+public:
+
+    /**
+     * Default constructor.
+     *
+     * @param explicitUpdateIfSubscriptionUnavailable In case there is no subscription response
+     * for a certain type of participant it will be replaced with an explicit request.
+     */
+    TraCISubscriptionManager(bool explicitUpdateIfSubscriptionUnavailable = true);
+
+    /**
+     * Default destructor.
+     */
+    virtual ~TraCISubscriptionManager() = default;
```

Comment:
This class seems like it is not intended to be used polymorphically, so you can just omit this.